docs/architecture/api.md

```diff
@@ -0,0 +1,43 @@
+# API Design
+
+Our API was designed with three principles in mind:
+
+1. All interactions of user's data should be done through the API, so that
+there is a consistent interface across the tools (CLI, Web UI, etc.).
+1. Resources should be ReST-ful and operations should be atomic so that intent
+is clear and human readable.
+1. API should be versioned and self-documented, so that client code generation
```

Comment:
Should this document state the current version that the API is on. And instead of referencing /v3 everywhere, we can just reference the endpoint and state that it is prefixed with /v3 ? 

It might make updating the documentation easier when we update the version of the API